### PR TITLE
storage: Install VDO support on demand 

### PR DESCRIPTION
**Pull request by mvollmer**
- [x] #8913 
- [x] #9372

### CODE REVIEW

File: pkg/storaged/client.js [hunk 1]
```diff
@@ -434,15 +434,7 @@
...
-                                   "Get", [ "org.freedesktop.PackageKit", "VersionMajor" ]).then(
-                                       function () {
-                                           client.features.packagekit = true;
-                                           return cockpit.resolve();
-                                       },
-                                       function () {
-                                           return cockpit.resolve();
-                                       });
+                    return cockpit.resolve();
                 });
```
Comment (martinpitt) [inline pkg/storaged/client.js]: The function wrapper is not needed any more here, is it? I. e. just `.then(..., cockpit.resolve);` ?

File: pkg/storaged/optional-panel.jsx [hunk 2]
```diff
@@ -87,6 +87,46 @@ function install_dialog(data, action) {
...
+export function optional_panel_init() {
+    return PK.call("/org/freedesktop/PackageKit", "org.freedesktop.DBus.Properties",
+                   "Get", [ "org.freedesktop.PackageKit", "VersionMajor" ]).then(
+        function () {
+            packagekit_available = true;
+        },
+        function () {
+            packagekit_available = false;
+        });
+}
```
Comment (martinpitt) [inline pkg/storaged/optional-panel.jsx]: When you move it around already, why not move this into pkg/lib/packagekit.es6?

File: pkg/storaged/vdos-panel.jsx [hunk 3]
```diff
@@ -155,15 +155,22 @@ export class VDOsPanel extends React.Component {
         );
 
         var vdo_feature = {
-            is_enabled: () => client.features.vdo
+            is_enabled: () => client.features.vdo,
+            packages: get_config("vdo_packages", false),
+            enable: () => {
+                client.features.vdo = true;
+                client.vdo_overlay.start();
+            }
```
Comment (martinpitt) [inline pkg/storaged/vdos-panel.jsx]: This looks really nice now!

**Comment by mvollmer** [comment]
> Do you think some of the OptionalPanel refactoring should already get moved into #8913, or even into its own PR? Or do you still want to experiment with that?

It should now go to #8913.  The code will move quite a bit when all the progress happens in the dialog.

**Comment by martinpitt** [comment]
Looks like the VDO failures on rhel-7 and centos-7 are real.

**Comment by mvollmer** [comment]
> Looks like the VDO failures on rhel-7 and centos-7 are real.

Cockpit is erroneously configured to install VDO on demand on rhel-7 and centos-7.  We only want that on rhel-7-5 and rhel-8.

**Comment by mvollmer** [comment]
> We only want that on rhel-7-5 and rhel-8.

I didn't find a way to distinguish rhel-7 from rhel-7-5 in cockpit.spec.  Should we actually install vdo in the rhel-7 image?  Can we?

**Comment by mvollmer** [comment]
> Cockpit is erroneously configured to install VDO on demand on rhel-7 and centos-7. We only want that on rhel-7-5 and rhel-8.

I was confused, sorry.  Let's get the vdo packages into all the images that we can and then see how to configure Cockpit here.  I guess that we can have vdo also for centos-7 and rhel-7.

**Comment by mvollmer** [comment]
> Let's get the vdo packages into all the images that we can

#9372

**Comment by mvollmer** [comment]
We can enable "vdo on demand" on centos-7, rhel-7, rhel-7-5, and rhel-x.

**Comment by mvollmer** [comment]
Added a test.

File: test/verify/check-storage-vdo [hunk 4]
```diff
@@ -261,5 +262,33 @@ config: !Configuration
...
+        b.click("button:contains('Install VDO support')")
+
+        self.dialog_wait_open()
+        b.wait_in_text("#dialog", "vdo will be installed")
+        b.wait_in_text("#dialog", "Total size:")
+        self.dialog_apply()
+        self.dialog_wait_close()
+
+        b.wait_present("#vdos button .fa-plus")
+        b.snapshot("XXX")
```
Comment (martinpitt) [inline test/verify/check-storage-vdo]: debugging leftover

**Comment by martinpitt** [comment]
Force-pushed to drop the debugging leftover, rebase to current master, and add a Closes #. If this passes tests, LGTM.

**Comment by martinpitt** [comment]
New test fails on rhel-x, looking.

**Comment by martinpitt** [comment]
PackageKit is still [broken](https://bugzilla.redhat.com/show_bug.cgi?id=1600452) on rhel-x, and on top of that `createrepo_c` does not exist any more. So I blacklisted the test just like the PCP one.

**Pull request merged**